### ISSUE TITLE
Warning: This shell is running on your local machine

### ISSUE THREAD

**Issue by seyeeet** [microsoft/vscode-remote-release#5055]
I am running vscode on a server from my local machine via ssh-remote
I recently start getting the warning that `Warning: This shell is running on your local machine`,
should I be worry about this warning? should I do something about it?

**Comment by adrianseidel** [comment]
I have the same warning and all things which are related to web like get, post, ... doesnt work anymore

**Comment by yipliu** [comment]
The same issue to me!

**Comment by mwyr** [comment]
Same for me. I run vscode-insiders on laptop (win10) to connect to development platform on RHEL8 with remote-ssh. When I tried last Friday (May, 14th) everything worked well.

```
Warning: This shell is running on your local machine

/usr/bin/bash: : No such file or directory.
```



VSCode About:
```
[Window Title]
Visual Studio Code - Insiders

[Main Instruction]
Visual Studio Code - Insiders

[Content]

Version: 1.57.0-insider (user setup)
Commit: 40d5e6796fbc32d67dd0009e5b0027003803fd7e
Date: 2021-05-14T20:13:58.234Z (2 days ago)
Electron: 12.0.7
Chrome: 89.0.4389.128
Node.js: 14.16.0
V8: 8.9.255.25-electron.0
OS: Windows_NT x64 10.0.19041
```

**Comment by massile** [comment]
Same problem here, I used to be able to launch a task from a remote machine (via SSH) last week.
But now I get this error `/usr/bin/zsh: : No such file or directory.`

VSCode about:

```
Version: 1.57.0-insider
Commit: 40d5e6796fbc32d67dd0009e5b0027003803fd7e
Date: 2021-05-14T20:14:19.367Z
Electron: 12.0.7
Chrome: 89.0.4389.128
Node.js: 14.16.0
V8: 8.9.255.25-electron.0
OS: Linux x64 5.8.0-53-generic snap
```

**Comment by DrShushen** [comment]
Same issue, I briefly see the message:
```
Warning: This shell is running on your local machine

/usr/bin/bash: : No such file or directory.
```
when trying to run VSCode debugger while connected to remote machine (SSH). 
The debugger then fails to launch.

VSCode Insiders about:
```
Version: 1.57.0-insider (user setup)
Commit: 82767cc1d7bf8cdea0f2897276d5d15aee91f3d9
Date: 2021-05-17T05:15:14.544Z
Electron: 12.0.7
Chrome: 89.0.4389.128
Node.js: 14.16.0
V8: 8.9.255.25-electron.0
OS: Windows_NT x64 10.0.21382
```
Remote development plugin versions:
```
Remote Development: v0.21.0
Remote - SSH: 0.65.4
```
Possibly relevant settings:
```
"terminal.integrated.inheritEnv": true
```
but tried also with (`false`) with same result.

---

However, this does **not** happen and the debugger launches correctly under the exact same setup but in my non-Insiders VSCode:
```
Version: 1.56.2 (user setup)
Commit: 054a9295330880ed74ceaedda236253b4f39a335
Date: 2021-05-12T17:13:13.157Z
Electron: 12.0.4
Chrome: 89.0.4389.114
Node.js: 14.16.0
V8: 8.9.255.24-electron.0
OS: Windows_NT x64 10.0.21382
```
Remote development plugin versions same as above.

**Comment by yipliu** [comment]
@DrShushen @massile @mwyr  This is so weird! The issue disappeared after restarting the Server.

**Comment by KiruyaMomochi** [comment]
Same issue here.
![image](https://user-images.githubusercontent.com/65301509/118604306-97615980-b7e7-11eb-9f2a-f20f0fd5b8cd.png)

### What I am doing?
Debug a c++ program.

### What's my environment?
Version: 1.57.0-insider (user setup)
Commit: 45aafeb326d0d3d56cbc9e2932f87e368dbf652d
Date: 2021-05-18T05:14:15.253Z
Electron: 12.0.7
Chrome: 89.0.4389.128
Node.js: 14.16.0
V8: 8.9.255.25-electron.0
OS: Windows_NT x64 10.0.21382
ms-vscode-remote.vscode-remote-extensionpack: v0.21.0
ms-vscode-remote.remote-wsl: v0.56.2
ms-vscode-remote.remote-ssh: v0.65.4

### `launch.json`
```json
{
    "name": "(gdb) Launch",
    "type": "cppdbg",
    "request": "launch",
    "program": "${workspaceRoot}/src/a",
    "args": [],
    "stopAtEntry": false,
    "cwd": "${workspaceFolder}/src",
    "environment": [],
    "externalConsole": false,
    "MIMode": "gdb",
    "setupCommands": [
        {
            "description": "Enable pretty-printing for gdb",
            "text": "-enable-pretty-printing",
            "ignoreFailures": true
        }
    ]
}
```

**Comment by massile** [comment]
@DrShushen 
> but tried also with (`false`) with same result.
> 
> However, this does **not** happen and the debugger launches correctly under the exact same setup but in my non-Insiders 

I confirm that this issue is only happening on Insiders, it seems to work properly on non-Insiders VSCode (I still have a warning but the SSH connection works).

I'm switching for now,

**Comment by DrShushen** [comment]
@yipliu Restarting my VM or removing `.vscode-server-insiders/` then reconnecting doesn't seem to have fixed this issue for me.

**Comment by wilbur4321** [comment]
I'm seeing the same thing, but using this, instead of the normal `integratedTerminal`, does seem to solve it:
```
"console": "internalConsole"
```

Having problems with remote path mapping, however.

**Comment by codsane** [comment]
> @DrShushen
> I confirm that this issue is only happening on Insiders, it seems to work properly on non-Insiders VSCode (I still have a warning but the SSH connection works).
> 
> I'm switching for now,

Not the case here, issue is occurring regardless of VSCode build.

**Comment by Tyriar** [comment]
This warning is typically nothing to worry about, it's been added to help differentiate local from remote terminals when in remote workspaces. It should only be showing up when you run the `Create New Integrated Terminal (Local)` command though, is this only happening when debugging?

**Comment by seyeeet** [comment]
@Tyriar I see that when I am not debugging as well. typically when I am connecting via ssh-remote to a server

**Comment by navneetkarnani** [comment]
I see this error when opening the Javascript Debugger terminal on a Remote - Container connected environment.

**Comment by rzhao271** [comment]
@seyeeet is the warning still showing up for you?

**Comment by seyeeet** [comment]
@rzhao271 It was with 1.56.1, but I cannot update my vscode since I am traveling now, so I dont know tbh

**Comment by seyeeet** [comment]
@rzhao271 I updated my vscode and extensions and I still see the warning.
Here is what im using now:
```
Version: 1.56.2 (user setup)
Date: 2021-05-12T17:13:13.157Z
Electron: 12.0.4
Chrome: 89.0.4389.114
Node.js: 14.16.0
V8: 8.9.255.24-electron.0
OS: Windows_NT x64 10.0.18363
```

**Comment by navneetkarnani** [comment]
I see this too. When I start the Javascript Debug Terminal, within a remove-container, not when I start the "standard" bash terminal.

**Comment by Tyriar** [comment]
The milestone is May so this will be fixed in v1.57, you can test it using the [insiders build](https://code.visualstudio.com/insiders).
